packages/governance/src/multisig/multisig.cairo

```diff
@@ -0,0 +1,646 @@
+            salt: felt252
+        ) {
+            let call = Call { to, selector, calldata };
+            self.execute_transaction_batch(array![call].span(), salt)
+        }
+
+        /// Executes a confirmed batch transaction.
+        ///
+        /// Requirements:
+        ///
+        /// - The caller must be a registered signer.
+        /// - The transaction must be confirmed and not yet executed.
+        ///
+        /// Emits a `TransactionExecuted` event.
+        fn execute_transaction_batch(
+            ref self: ComponentState<TContractState>, calls: Span<Call>, salt: felt252
+        ) {
+            let id = self.hash_transaction_batch(calls, salt);
+            match self.resolve_tx_state(id) {
+                TransactionState::NotFound => panic_with_felt252(Errors::TX_NOT_FOUND),
+                TransactionState::Pending => panic_with_felt252(Errors::TX_NOT_CONFIRMED),
+                TransactionState::Executed => panic_with_felt252(Errors::TX_ALREADY_EXECUTED),
+                TransactionState::Confirmed => {
+                    let caller = starknet::get_caller_address();
+                    self.assert_one_of_signers(caller);
+                    let mut tx_info = self.Multisig_tx_info.read(id);
+                    tx_info.is_executed = true;
+                    self.Multisig_tx_info.write(id, tx_info);
+                    for call in calls {
+                        let Call { to, selector, calldata } = *call;
+                        call_contract_syscall(to, selector, calldata).unwrap_syscall();
+                    };
+                    self.emit(TransactionExecuted { id });
+                }
+            };
+        }
+
+        /// Returns the computed identifier of a transaction containing a single call.
+        fn hash_transaction(
+            self: @ComponentState<TContractState>,
+            to: ContractAddress,
+            selector: felt252,
+            calldata: Span<felt252>,
+            salt: felt252
+        ) -> TransactionID {
+            let call = Call { to, selector, calldata };
+            self.hash_transaction_batch(array![call].span(), salt)
+        }
+
+        /// Returns the computed identifier of a transaction containing a batch of calls.
+        fn hash_transaction_batch(
+            self: @ComponentState<TContractState>, calls: Span<Call>, salt: felt252
+        ) -> TransactionID {
+            PedersenTrait::new(0).update_with(calls).update_with(salt).finalize()
+        }
+    }
+
+    #[generate_trait]
+    pub impl InternalImpl<
+        TContractState, +HasComponent<TContractState>, +Drop<TContractState>
+    > of InternalTrait<TContractState> {
+        /// Initializes the Multisig component with the initial `quorum` and `signers`.
+        /// This function must be called during contract initialization to set up the initial state.
+        ///
+        /// Requirements:
+        ///
+        /// - The function must be called only once.
```

Comment:
```suggestion
```

For consistency. People should be familiar with the pattern since we are using this in other components without the comment.